diesel/src/connection/sqlite/stmt.rs

```diff
@@ -0,0 +1,141 @@
+            error => Err(DatabaseError(super::error_message(error).into()))
+        }
+    }
+
+    pub fn bind(&mut self, tpe: SqliteType, value: Option<Vec<u8>>) -> QueryResult<()> {
+        self.bind_index += 1;
+        let result = unsafe { match (tpe, value) {
+            (_, None) =>
+                ffi::sqlite3_bind_null(self.inner_statement, self.bind_index),
+            (SqliteType::Binary, Some(bytes)) =>
+                ffi::sqlite3_bind_blob(
+                    self.inner_statement,
+                    self.bind_index,
+                    bytes.as_ptr() as *const libc::c_void,
+                    bytes.len() as libc::c_int,
+                    ffi::SQLITE_TRANSIENT(),
+                ),
+            (SqliteType::Text, Some(bytes)) =>
+                ffi::sqlite3_bind_text(
+                    self.inner_statement,
+                    self.bind_index,
+                    bytes.as_ptr() as *const libc::c_char,
+                    bytes.len() as libc::c_int,
+                    ffi::SQLITE_TRANSIENT(),
+                ),
+            (SqliteType::Float, Some(bytes)) => {
+                let value = try!(bytes.as_slice().read_f32::<BigEndian>()
+                    .map_err(|e| QueryBuilderError(Box::new(e))));
+                ffi::sqlite3_bind_double(
+                    self.inner_statement,
+                    self.bind_index,
+                    value as libc::c_double,
+                )
+            }
+            (SqliteType::Double, Some(bytes)) => {
+                let value = try!(bytes.as_slice().read_f64::<BigEndian>()
+                    .map_err(|e| QueryBuilderError(Box::new(e))));
+                ffi::sqlite3_bind_double(
+                    self.inner_statement,
+                    self.bind_index,
+                    value as libc::c_double,
+                )
+            }
+            (SqliteType::SmallInt, Some(bytes)) => {
+                let value = try!(bytes.as_slice().read_i16::<BigEndian>()
+                    .map_err(|e| QueryBuilderError(Box::new(e))));
+                ffi::sqlite3_bind_int(
+                    self.inner_statement,
+                    self.bind_index,
+                    value as libc::c_int,
+                )
+            }
+            (SqliteType::Integer, Some(bytes)) => {
+                let value = try!(bytes.as_slice().read_i32::<BigEndian>()
+                    .map_err(|e| QueryBuilderError(Box::new(e))));
+                ffi::sqlite3_bind_int(
+                    self.inner_statement,
+                    self.bind_index,
+                    value as libc::c_int,
+                )
+            }
+            (SqliteType::Long, Some(bytes)) => {
+                let value = try!(bytes.as_slice().read_i64::<BigEndian>()
+                    .map_err(|e| QueryBuilderError(Box::new(e))));
+                ffi::sqlite3_bind_int64(
+                    self.inner_statement,
+                    self.bind_index,
+                    value,
```

Comment:
I feel like there's a lot of duplication here, which is causing my previous "arcane" comments, I wonder if we can pull some of that out?